content/param_id_func/contents.lr

```diff
@@ -0,0 +1,471 @@
+name: ids for fixtures and parametrize
+---
+author: hackebrot
+---
+content:
+
+# parameters for tests
+
+**pytest** comes with a handful of powerful tools to generate parameters for a
+test, so you can run various scenarios against the same test implementation.
+
+- ``params`` on a ``@pytest.fixture``
+- ``parametrize`` marker
+- ``pytest_generate_tests`` hook with ``metafunc.parametrize``
+
+All of the above have their individual strengths and weaknessses. In this post
+I'd like to cover ids for tests and why I think it's a good idea to use them.
+
+Please check out older posts on this blog to find out more about
+parametrization in pytest, if you haven't already.
+
+- [create tests via parametrization](http://hackebrot.github.io/pytest-tricks/create_tests_via_parametrization/)
+- [mark.parametrize with indirect](http://hackebrot.github.io/pytest-tricks/mark_parametrize_with_indirect/)
+
+# test items
+
+To understand how **pytest** generates ids for tests, we first need to know
+what test items are.
+
+For **pytest** a test item is a single test, defined by an underlying test
+function along with its setup and teardown code. On top of that all of the
+fixtures used by the test function make up a test item. Essentially, everything
```

Comment:
I find it a little weird you discuss setup/teardown code and fixtures separately.  The fixtures are the setup and teardown normally in my view.  I'd maybe phrase it as "an underlying test function along with the setup and teardown code for the fixtures it uses."  There are probably other variations.